session_test.go

```diff
@@ -0,0 +1,107 @@
+package mode
+
+import (
+	"fmt"
+	"io"
+	//	"io/ioutil"
+	"net"
+	"os"
+	"sync"
+	"testing"
+
+	packet "github.com/moderepo/device-sdk-go/mqtt_packet"
+	"github.com/stretchr/testify/assert"
+)
+
+var testPort int = 9999
+var wg sync.WaitGroup
+var resultingPubMessage string
+var resultingPubTopic string
+
+func ReadStream(conn net.Conn) bool {
```

Comment:
This method is not need exported names.
```suggestion
func readStream(conn net.Conn) bool {
```

Reply:
Useful test helper function... nice.